x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -167,6 +167,30 @@ public void startDeployment(TrainedModelDeploymentTask task, ActionListener<Trai
         ActionListener<GetTrainedModelsAction.Response> getModelListener = ActionListener.wrap(getModelResponse -> {
             assert getModelResponse.getResources().results().size() == 1;
             TrainedModelConfig modelConfig = getModelResponse.getResources().results().get(0);
+            String modelId = modelConfig.getModelId();
+            String platformArchitecture = modelConfig.getPlatformArchitecture();
+
+            ActionListener<Void> failureListener = new ActionListener<Void>() {
+
+                @Override
+                public void onResponse(Void o) {
+
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    failedDeploymentListener.onFailure(e);
+                }
+            };
+
+            MlPlatformArchitecturesUtil.verifyMlNodesAndModelArchitectures(
+                failureListener,
+                client,
+                threadPool,
+                platformArchitecture,
+                modelId
+            );
```

Comment:
This needs moving to a separate listener at the same level as `getModelListener`.

What was `getModelListener` should become `verifyArchitectureListener`, and should be called from a successful reponse to `verifyMlNodesAndModelArchitectures`.

Reply:
I am happy to move the `faillureListener` up to the same level as `getModelListener`, but I am confused about your other suggestions.

Is it the `getModelListener` that should become `verifyArchitectureListener` or is it the `failureListener` that should be come `verifyArchitectureListener`?

The `getModelListener` is called from `executeAsyncWithOrigin` below, so I am not sure why that should change. We are currently getting the model config from that `executeAsyncWithOrigin`.

Shouldn't we be calling `verifyMlNodesAndModelArchitectures` (with `verifyArchitectureListener`) from the existing `getModelListener`?

Reply:
Yes, we need a chain of steps.

Get models -> verify models -> the next thing

Currently `getModelListener` does "the next thing.

Instead `getModelListener` needs to do the verification, and then a new listener needs to do "the next thing".

So move all the code that used to be in `getModelListener` into a new listener called `verifyModelListener` declared immediately above. Then you want to pass on the `getModelResponse` to that in the event that verification succeeds.